app/src/main/java/com/sun/basic_japanese/kanji_basic/KanjiBasicFragment.kt

```diff
@@ -0,0 +1,114 @@
+import android.view.ViewGroup
+import com.sun.basic_japanese.R
+import com.sun.basic_japanese.base.BaseFragment
+import com.sun.basic_japanese.base.FragmentInteractionListener
+import com.sun.basic_japanese.data.model.KanjiBasic
+import com.sun.basic_japanese.data.repository.KanjiRepository
+import com.sun.basic_japanese.data.source.local.AppDatabase
+import com.sun.basic_japanese.data.source.local.KanjiLocalDataSource
+import com.sun.basic_japanese.data.source.remote.KanjiRemoteDataSource
+import com.sun.basic_japanese.util.Extensions.showToast
+import kotlinx.android.synthetic.main.bottom_navigation.*
+import kotlinx.android.synthetic.main.fragment_kanji.*
+
+class KanjiBasicFragment @SuppressLint("ValidFragment") private constructor() : BaseFragment(),
+    KanjiBasicContract.View,
+    KanjiBasicRecyclerAdapter.OnKanjiBasicItemClickListener {
+
+    private var listener: OnKanjiBasicFragmentInteractionListener? = null
+    private var kanjiBasicPresenter: KanjiBasicContract.Presenter? = null
+    private var currentLesson = 1
+    private val kanjiBasicRecyclerAdapter = KanjiBasicRecyclerAdapter(this)
+
+    override fun onAttach(context: Context?) {
+        super.onAttach(context)
+        if (context is OnKanjiBasicFragmentInteractionListener) {
+            listener = context
+        } else {
+            throw RuntimeException("${context.toString()} $ERROR_IMPLEMENT_FRAGMENT_INTERACTION_LISTENER")
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_kanji, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initPresenter()
+        initView()
+        kanjiBasicPresenter?.getKanjiBasicData(currentLesson)
+    }
+
+    override fun showKanjiBasicData(kanjiBasicList: List<KanjiBasic>) {
+        kanjiBasicRecyclerAdapter.setKanjiBasicList(kanjiBasicList)
+    }
+
+    override fun showToast(message: String) {
+        context?.showToast(message)
+    }
+
+    override fun showKanjiBasicDetail(kanjiBasic: KanjiBasic) {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    private fun initPresenter() {
+        context?.let {
+            val local = KanjiLocalDataSource.getInstance(AppDatabase.getInstance(it))
+            val remote = KanjiRemoteDataSource.getInstance()
+            val kanjiRepository =
+                KanjiRepository.getInstance(local, remote)
+            kanjiBasicPresenter = KanjiBasicPresenter(this, kanjiRepository)
+        }
+    }
+
+    private fun initView() {
+        recyclerKanjiBasic.apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = kanjiBasicRecyclerAdapter
+        }
+
+        buttonLessonSelect.setOnClickListener {
```

Comment:
chỉ nên bắt sự kiện ở đây, việc xử lý thì viết trong hàm onClick() 
```suggestion
        buttonLessonSelect.setOnClickListener(this)
```